### PR TITLE
[border-router] add random delay before changing routing policies

### PR DESCRIPTION
**Pull request by wgtdkp**
We also fixed an issue that a border router may adds the OMR prefix of current network as an external route.

### CODE REVIEW

File: src/core/border_router/routing_manager.hpp [hunk 1]
```diff
@@ -296,6 +301,10 @@ class RoutingManager : public InstanceLocator
 
     TimerMilli mRouterSolicitTimer;
     uint8_t    mRouterSolicitCount;
+
+    // The timer used to delay routing policy evaluation with
+    // random delay between 0 and `kMaxRoutingPolicyDelay`.
```
Comment (abtink) [inline src/core/border_router/routing_manager.hpp:306]: nit: I think we can consider removing this comment?

**Review by abtink — APPROVED** [review]
Looks great. Thanks. 👍

File: src/core/border_router/routing_manager.hpp (hunk 1 above)
Comment (wgtdkp) [reply src/core/border_router/routing_manager.hpp]: sure, updated.

File: src/core/border_router/routing_manager.hpp [hunk 2]
```diff
@@ -170,6 +170,7 @@ class RoutingManager : public InstanceLocator
         kMaxRaDelayTime              = 500, // The maximum delay of sending RA after receiving RS. In milliseconds.
         kRtrSolicitationInterval     = 4,   // The interval between Router Solicitations. In seconds.
         kMaxRtrSolicitationDelay     = 1,   // The maximum delay for initial solicitation. In seconds.
+        kMaxRoutingPolicyDelay       = 1,   // The maximum delay for routing policy evaluation. In seconds.
```
Comment (simonlingoogle) [inline src/core/border_router/routing_manager.hpp:173]: 1s is too quick ?

```suggestion
        kMaxRoutingPolicyDelay       = 3,   // The maximum delay for routing policy evaluation. In seconds.
```

File: tests/scripts/thread-cert/node.py [hunk 3]
```diff
@@ -137,6 +137,7 @@ def _launch_docker(self):
 
         cmd = f'docker exec -i {self._docker_name} ot-ctl'
         self.pexpect = pexpect.popen_spawn.PopenSpawn(cmd, timeout=10)
+        self.pexpect.logfile = sys.stdout.buffer
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/node.py:140]: We already do this if `VERBOSE=1`

File: src/core/border_router/routing_manager.cpp [hunk 4]
```diff
@@ -626,6 +627,17 @@ void RoutingManager::EvaluateRoutingPolicy(void)
...
 
+void RoutingManager::StartEvaluateRoutingPolicy(void)
+{
+    uint32_t randomDelay;
+
+    static_assert(kMaxRoutingPolicyDelay > 0, "invalid maximum routing policy evaluation delay");
+    randomDelay = Random::NonCrypto::GetUint32InRange(0, Time::SecToMsec(kMaxRoutingPolicyDelay));
+
+    otLogInfoBr("start evaluating routing policy, scheduled in %u milliseconds", randomDelay);
+    mRoutingPolicyTimer.Start(randomDelay);
```
Comment (simonlingoogle) [inline src/core/border_router/routing_manager.cpp:638]: Need to stop the Timer in RoutingManager::Stop ?

File: tests/scripts/thread-cert/node.py (hunk 3 above)
Comment (wgtdkp) [reply tests/scripts/thread-cert/node.py:140]: Thanks, not aware of that, will revert.

File: src/core/border_router/routing_manager.hpp (hunk 2 above)
Comment (wgtdkp) [reply src/core/border_router/routing_manager.hpp:173]: When the Routing Manager is started or all existing on-link prefixes are expired, we start
sending router solicitation in a random delay between 0 and 1 seconds. We are trying to use
the scale with it.

File: src/core/border_router/routing_manager.cpp [hunk 5]
```diff
@@ -316,7 +319,7 @@ void RoutingManager::HandleNotifierEvents(Events aEvents)
 
     if (aEvents.Contains(kEventThreadNetdataChanged))
     {
-        EvaluateRoutingPolicy();
+        StartEvaluateRoutingPolicy();
```
[the remark this answers is not in the document]
Comment (wgtdkp) [reply src/core/border_router/routing_manager.cpp:322]: SGTM!

File: src/core/border_router/routing_manager.cpp [hunk 6]
```diff
@@ -1155,6 +1174,29 @@ bool RoutingManager::AddDiscoveredPrefix(const Ip6::Prefix &aPrefix,
...
 
+bool RoutingManager::NetworkDataContainsOmrPrefix(const Ip6::Prefix &aPrefix) const
+{
+    NetworkData::Iterator           iterator = NetworkData::kIteratorInit;
+    NetworkData::OnMeshPrefixConfig onMeshPrefixConfig;
+    bool                            contain = false;
+
+    while (Get<NetworkData::Leader>().GetNextOnMeshPrefix(iterator, onMeshPrefixConfig) == OT_ERROR_NONE)
+    {
+        if (!IsValidOmrPrefix(onMeshPrefixConfig.GetPrefix()) || !onMeshPrefixConfig.mSlaac || onMeshPrefixConfig.mDp)
```
Comment (simonlingoogle) [inline src/core/border_router/routing_manager.cpp:1185]:
```suggestion
        if (!IsValidOmrPrefix(onMeshPrefixConfig))
```

This condition appeared more than once. Define a new method to encapsulate the logic?

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1: 

with a nit

**Review by jwhui — APPROVED** [review]

**Pull request merged**